src/components/OptionsSelector.js

```diff
@@ -123,92 +221,93 @@ class OptionsSelector extends Component {
+            focusedIndex: newFocusedIndex,
+        });
 
-        if (this.props.disableArrowKeysActions && e.nativeEvent.key.startsWith('Arrow')) {
+        if (newOptions.length <= newFocusedIndex) {
             return;
         }
+        this.scrollToIndex(newFocusedIndex);
+    }
 
-        switch (e.nativeEvent.key) {
-            case 'Enter': {
-                this.selectRow(allOptions[this.state.focusedIndex]);
-                e.preventDefault();
-                break;
-            }
-
-            case 'ArrowDown': {
-                this.setState((prevState) => {
-                    let newFocusedIndex = prevState.focusedIndex + 1;
+    componentWillUnmount() {
+        if (this.unsubscribeEnter) {
+            this.unsubscribeEnter();
+        }
 
-                    // Wrap around to the top of the list
-                    if (newFocusedIndex > allOptions.length - 1) {
-                        newFocusedIndex = 0;
-                    }
+        if (this.unsubscribeCTRLEnter) {
+            this.unsubscribeCTRLEnter();
+        }
+    }
 
-                    const {index, sectionIndex} = allOptions[newFocusedIndex];
-                    this.scrollToFocusedIndex(sectionIndex, index);
-                    return {focusedIndex: newFocusedIndex};
+    /**
+     * Flattens the sections into a single array of options.
+     * Each object in this array is enhanced to have:
+     *
+     *   1. A `sectionIndex`, which represents the index of the section it came from
+     *   2. An `index`, which represents the index of the option within the section it came from.
+     *
+     * @returns {Array<Object>}
+     */
+    flattenSections() {
+        const allOptions = [];
+        _.each(this.props.sections, (section, sectionIndex) => {
+            _.each(section.data, (option, optionIndex) => {
+                allOptions.push({
+                    ...option,
+                    sectionIndex,
+                    index: optionIndex,
                 });
+            });
+        });
+        return allOptions;
+    }
 
-                e.preventDefault();
-                break;
-            }
+    /**
+     * @param {Number} index
+     */
+    updateFocusedIndex(index) {
+        this.setState({focusedIndex: index}, () => this.scrollToIndex(index));
+    }
 
-            case 'ArrowUp': {
-                this.setState((prevState) => {
-                    let newFocusedIndex = prevState.focusedIndex - 1;
+    /**
+     * Scrolls to the focused index within the SectionList
+     *
+     * @param {Number} index
+     */
+    scrollToIndex(index) {
+        const option = this.state.allOptions[index];
+        if (!this.list || !option) {
+            return;
+        }
 
-                    // Wrap around to the bottom of the list
-                    if (newFocusedIndex < 0) {
-                        newFocusedIndex = allOptions.length - 1;
-                    }
+        const {index: itemIndex, sectionIndex} = option;
```

Comment:
suggestion: use less fancy syntax here so we don't create friction for people who work on JS less frequently than us.
```suggestion
        const itemIndex = option.index;
        const sectionIndex = option.sectionIndex;
```